libraries/testing/tester.cpp

```diff
@@ -341,20 +342,38 @@ namespace eosio::testing {
       control->add_indices();
       control->testing_allow_voting(true);
       if (lambda) lambda();
+      [[maybe_unused]] auto block_start_connection = control->block_start().connect([](block_num_type num) {
+         // only block number is signaled, in forking tests will get the same block number more than once.
+      });
+      [[maybe_unused]] auto accepted_block_header_connection = control->accepted_block_header().connect([this](const block_signal_params& t) {
+            const auto& [block, id] = t;
+            FC_ASSERT(block);
+            auto i = blocks_signaled.find(id);
+            BOOST_TEST((i == blocks_signaled.end()), "should get accepted_block_header signal only once, and before accepted_block signal");
+            blocks_signaled[id] = block_signal::accepted_block_header;
+         });
       chain_transactions.clear();
-      [[maybe_unused]] auto accepted_block_connection = control->accepted_block().connect([this]( block_signal_params t ){
-        const auto& [ block, id ] = t;
-        FC_ASSERT( block );
-          for( auto receipt : block->transactions ) {
-              if( std::holds_alternative<packed_transaction>(receipt.trx) ) {
-                  auto &pt = std::get<packed_transaction>(receipt.trx);
+      [[maybe_unused]] auto accepted_block_connection = control->accepted_block().connect([this](const block_signal_params& t) {
+            const auto& [block, id] = t;
+            FC_ASSERT(block);
+            BOOST_TEST(block->block_num() > lib_number);
+            auto i = blocks_signaled.find(id);
```

Comment:
Small detail, for some reason I find `i` a stange name for an iterator. Would prefer `it`.